kong/core/balancer.lua

```diff
@@ -1,41 +1,224 @@
-local dns_client = require "resty.dns.client"
+local cache = require "kong.tools.database_cache"
+local pl_tablex = require "pl.tablex"
+local responses = require "kong.tools.responses"
+local singletons = require "kong.singletons"
+local dns_client = require "resty.dns.client"  -- due to startup/require order, cannot use the one from 'singletons' here
+local ring_balancer = require "resty.dns.balancer"
 
 local toip = dns_client.toip
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_DEBUG = ngx.DEBUG
 
--- looks up a balancer for the target.
--- @param target the table with the target details
--- @return balancer if found, or nil if not found, or nil+error on error
-local get_balancer = function(target)
-  return nil  -- TODO: place holder, forces dns use to first fix regression
+local empty = pl_tablex.readonly {}
```

Comment:
I didn't, but that's to be changed then. Shouldn't it be `EMPTY_T` then? allcaps?